otodom.py

```diff
@@ -0,0 +1,362 @@
+    longitude = html_parser.find(itemprop="longitude").attrs["content"]
+    return [latitude, longitude]
+
+
+def get_offer_details(html_parser):
+    f = html_parser.find(class_="text-details clearfix").text
+    return [{d.split(': ')[0].strip(): d.split(': ')[1].strip()} for d in f.split("\n") if not re.match(r'^\s*$', d)]
+
+
+def get_offer_information(url, context=None):
+    # geting response
+    response = get_response_for_url(url)
+    content = response.content
+    html_parser = BeautifulSoup(content, "html.parser")
+
+    # getting meta values
+    cookie = get_cookie_from(response)
+    csrf_token = get_csrf_token(content)
+    offer_id = context['offer_id']
+
+    # getting offer details
+    try:
+        phone_numbers = get_offer_phone_numbers(offer_id, cookie, csrf_token)
+    except KeyError as e:
+        # offer was not present any more
+        return {}
+
+    number_replace_dict = {u'\xa0': "", " ": "", "-": "", "+48": ""}
+    phone_numbers = sum([replace_all(num, number_replace_dict).split(".") for num in phone_numbers], [])
+
+    facebook_description = get_offer_facebook_description(html_parser)
+
+    ninja_pv = get_offer_ninja_pv(content)
+    price = ninja_pv["ad_price"]
+    currency = ninja_pv["price_currency"]
+    voivodeship = ninja_pv["region_name"]
+    city = ninja_pv["city_name"]
+    try:
+        district = ninja_pv["district_name"]
+    except KeyError as e:
+        district = ""
+    surface = ninja_pv["surface"]
+    rooms = ninja_pv["rooms"]
+    poster_type = ninja_pv["poster_type"]
+
+    floor = get_offer_floor(html_parser)
+    total_floors = get_offer_total_floors(html_parser)
+
+    apartment_details = get_offer_apartment_details(html_parser)
+
+    additional_assets = get_offer_additional_assets(html_parser)
+
+    description = get_offer_description(html_parser)
+
+    photos = get_offer_photos_links(html_parser)
+
+    coordinates = get_offer_geographical_coordinates(html_parser)
+
+    offer_details = get_offer_details(html_parser)
+
+    return {
+        'surface': surface,
+        'rooms': rooms,
+        'floor': floor,
+        'total_floors': total_floors,
+        'poster_type': poster_type,
+        'price': price,
+        'currency': currency,
+        'city': city,
+        'district': district,
+        'voivodeship': voivodeship,
+        'geographical_coordinates': coordinates,
+        'phone_numbers': phone_numbers,
+        'description': description,
+        'offer_details': offer_details,
+        'photo_links': photos,
+        'apartment_details': apartment_details,
+        'additional_assets': additional_assets,
+        'facebook_description': facebook_description,
+        'meta': {
+            'cookie': cookie,
+            'csrf_token': csrf_token,
+            **context,
+        }
+    }
+
+
+if __name__ == '__main__':
+
+    # the following commented dict contains every possible filter with examples of its values
+
+    # input_dict = {
```

Comment:
maybe move it to get_category docstring[1]


[1] - https://www.python.org/dev/peps/pep-0257/